synthetic_signwriting/synthetic_signwriting.py

```diff
@@ -1,37 +1,86 @@
 import random
 
+import numpy as np
 from pose_format import Pose
 
+from synthetic_signwriting.hands.hands import read_npy_file, prorate_hand
+from synthetic_signwriting.faces.faces import prorate_face, read_pkl_file
+from synthetic_signwriting.synthetic_pose_helper import randomize_hand_pose, create_base_frame, pose_transition
+
 
 class SyntheticSignWriting:
-    pose = Pose(header=None, body=None)
 
     def __init__(self):
-        self.key_frames = []
+        with open('data/base_pose.pose', 'rb') as file:
+            data_buffer = file.read()
+        pose = Pose.read(data_buffer)
+        middle = int(len(pose.body.data) / 2)
+
+        self.hand_data = read_npy_file('data/hands.npy')
+        self.face_data = read_pkl_file('data/faces.pkl')
+        self.pose = pose
+        self.static_position = [pose.body.data[middle + 2 * i][0].copy() for i in range(4)]
         self.hands = []
         self.faces = []
 
-    def add_keyframe(self, face=True, left_hand=True, right_hand=True):
+    def add_keyframe(self, face=None, left_hand=None, right_hand=None):
         # Create and configure a new frame
-        frame = self.pose.create_frame()
-        if face:
-            frame.set_face(...)
-        if left_hand:
-            frame.set_left_hand(...)
-        if right_hand:
-            frame.set_right_hand(...)
+        self.hands.append((left_hand, right_hand))
+        self.faces.append(face)
+
+    def add_random_keyframe(self):
+        # choose a random hand and face pose
+        right_hand = randomize_hand_pose(self.hand_data)  # choose a random hand pose
+        left_hand = randomize_hand_pose(self.hand_data)  # choose a random hand pose
+
+        # choose a random face pose
+        face_type = random.choice(list(self.face_data.keys()))
+        element = random.choice(list(self.face_data[face_type].keys()))
+        if element == "source":
+            face = self.face_data[face_type][element]
+        else:
+            cathartic = random.choice(list(self.face_data[face_type][element].keys()))
+            face = self.face_data[face_type][element][cathartic]
+
+        # add the keyframe
+        self.add_keyframe(face=face, left_hand=left_hand, right_hand=right_hand)
 
     def render_signwriting(self):
         # Use the list of hands and faces to generate SignWriting
         return "..."
 
-    def render_pose(self):
-        self._smooth()
-        # Apply camera transformation and render the pose sequence
-        return "..."
+    def render_pose(self, frame_suspension=100):
+        self.pose.body.data = None
+        zero_frame = np.zeros((frame_suspension - 2, 1, 543, 3))  # the -2 is to remove the first and last frame
```

Comment:
543 and 3 are hardcoded, but should not be
`_, _, points, dimensions = self.pose.body.data.shape`